### DISCUSSION TITLE
webRequest in persistent scripts.

### DISCUSSION THREAD

**Discussion by nexensys** [ScratchAddons/ScratchAddons#2588] [Help] [answered]
Hi, I was just wondering if there are any differences between using webrequest in an addon and using it in chrome. I am trying to make an addon that changes the scratchblocks in the forums to 3.0 style, but to do that I will have to cancel the request to scratch's servers...I know I have to include it in the `permissions` of my manifest, but is there anything else?

**Comment by nexensys** [comment, marked as answer]
Never mind, I can't use webrequest for this. I am going to have to un-parse the scratchblocks.

**Comment by WorldLanguages** [comment]
That addon was already tried multiple times... Do not create it without discussing first how it will work. Get into the development discord server if possible

**Reply by nexensys** [reply]
> That addon was already tried multiple times...

Yeah, I know I have tried myself before...I just gave up 20 minutes ago. I only tried again because I thought of webrequest, but it does not work the way I need it to.

> Get into the development discord server if possible

Sorry, I can't use discord.

**Reply by WorldLanguages** [reply]
> Sorry, I can't use discord.

lol but judging by you're username you're 21 y/o?

**Reply by nexensys** [reply]
XD nah, 16.

**Reply by nexensys** [reply]
Why, do I really sound that grown-up?

**Reply by cobaltt7** [reply]
ErrorGamer**2000**
Born in 2000

**Reply by nexensys** [reply]
Ah. No, I just chose that because it sounded better at the time and just 'ErrorGamer' was taken.

**Reply by GrahamSH-LLK** [reply]
But you can't use Discord at _16_? You should just get it and not tell your parents.

**Reply by nexensys** [reply]
And get killed. They have complete access to my computer, can see what apps I use, pages I visit, etc. *And* they don't trust discord. It **sucks**.

**Reply by Shluffy** [reply]
> And get killed. They have complete access to my computer, can see what apps I use, pages I visit, etc. _And_ they don't trust discord. It **sucks**.

family link over 13?

**Reply by WorldLanguages** [reply]
Well, you can show them this conversation. Your experience coding will be worse without Discord :P

**Reply by nexensys** [reply]
...Yeah they won't take that. They *really* don't trust it. I tried convincing them multiple times, and they said no every one of them. Also, they have done some digging and found a lot of bad stuff.

**Reply by GrahamSH-LLK** [reply]
Like what?

**Reply by nexensys** [reply]
I don't remember exactly, nut enough that they won't let me near it and it is blocked on the wifi...

**Reply by retronbv** [reply]
# [shameless plug]
maybe try using Ozark O.O
my cool unblocker [here](https://ozark.retron.gq)
Just click discord in the quick links section.

**Reply by nexensys** [reply]
Oh, SWEET thanks!

**Reply by retronbv** [reply]
np :) use it if you want

**Reply by nexensys** [reply]
oh, and btw scratch is blocked too but I foud a way around that...XD
